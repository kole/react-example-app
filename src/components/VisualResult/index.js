import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Visual = styled.div`
    display: flex;
    justify-content: flex-start;

    i.right {
        color: green
    }
    i.wrong {
        color: red
    }
`;

const VisualResult = (props) => {
    let { correctAnswer, givenAnswer } = props;
    correctAnswer = correctAnswer.toLowerCase();
    givenAnswer = givenAnswer.toString().toLowerCase();

    return (
        <Visual>
            { correctAnswer === givenAnswer ?
                <i className="material-icons right">check</i> :
                <i className="material-icons wrong">close</i>
            }
        </Visual>
    );
}

VisualResult.propTypes = {
    correctAnswer: PropTypes.string.isRequired,
    givenAnswer: PropTypes.bool.isRequired
}

export default VisualResult;
