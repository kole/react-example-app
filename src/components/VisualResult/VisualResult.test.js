/* eslint-env jest */

import React from 'react';
import renderer from 'react-test-renderer';
import VisualResult from './index.js';

describe('VisualResult', () => {
    test('Snapshot with right answer', () => {
        const component = renderer
            .create(<VisualResult correctAnswer="True" givenAnswer={true} />)
            .toJSON();
        expect(component).toMatchSnapshot();
    });

    test('Snapshot with wrong answer', () => {
        const component = renderer
            .create(<VisualResult correctAnswer="True" givenAnswer={false} />)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
});
