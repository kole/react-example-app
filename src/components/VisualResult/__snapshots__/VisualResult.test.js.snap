// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisualResult Snapshot with right answer 1`] = `
<div
  className="sc-bdVaJa KMFfT"
>
  <i
    className="material-icons right"
  >
    check
  </i>
</div>
`;

exports[`VisualResult Snapshot with wrong answer 1`] = `
<div
  className="sc-bdVaJa KMFfT"
>
  <i
    className="material-icons wrong"
  >
    close
  </i>
</div>
`;
